dlt/extract/utils.py

```diff
@@ -119,6 +119,25 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_generator(f: Any) -> Any:
+    """Wraps an async generator into a list with one awaitable"""
+
+    async def run() -> List[TDataItem]:
+        result: List[TDataItem] = []
+        try:
+            item: TDataItems = None
+            while item := await f.__anext__():
```

Comment:
the problem with this thing is that it will consume the whole generator in memory and then return results once. so ie. when there's a transformer it will receive a full dataset. right?

maybe we need to send each f.__next__() to a future pool and wrap this in a function that returns both a next result (after await) and the async iterator itself? so you put it into futures pool again and return the current value